roles/provision-unifiedpush-apb/tasks/provision-ups.yml

```diff
@@ -94,31 +100,33 @@
   shell: "oc get routes ups -n {{ namespace }} | grep -v NAME | awk '{print $2}'"
   register: ups_route
 
-- name: "Create ups secret yaml file"
-  template:
-    src: secret.yml.j2
-    dest: /tmp/secret.yaml
-
-- name: "Create UPS secret"
-  shell: "oc create -f /tmp/secret.yaml -n {{ namespace }}"
-
-- name: "Delete UPS Secret Template File"
-  file: path=/tmp/secret.yaml state=absent
-
 # We need to wait for the deployment to be ready before we can exit the ansible job
 - name: "Wait for all UPS containers to become ready"
   shell: oc get pods --namespace={{ namespace }} --selector="deploymentconfig=ups" -o jsonpath='{.items[*].status.containerStatuses[?(@.ready==true)].ready}'| wc -w
   register: ups_result
-  until: ups_result.stdout.find("2") != -1
+  until: ups_result.stdout.find("3") != -1
   retries: 30
   delay: 5
 
-- name: Create {{ namespace }} PushApplication on service host {{ unifiedpush_service.service.spec.cluster_ip }}
+- name: "Create {{ namespace }} PushApplication on service host {{ unifiedpush_service.service.spec.cluster_ip }}"
   uri:
     url: http://{{ unifiedpush_service.service.spec.cluster_ip }}/rest/applications
     method: POST
     body: '{"name" : "{{ namespace }}", "description" :  "Push application for {{ namespace }}"}'
     validate_certs: no
     body_format: json
     status_code: 201
+  retries: 5
```

Comment:
did it time out for you, @pb82  ?

Reply:
I've once seen it return a non-201 response and the APB fail with an error. Thought a few retries can't hurt.